packages/build/src/tarball.spec.ts

```diff
@@ -0,0 +1,133 @@
+          to.equal('mongosh-1.0.0-win32.zip');
+      });
+    });
+
+    context('when the build variant is macos', () => {
+      it('returns the tarball name', () => {
+        expect(tarballPath('', BuildVariant.MacOs, '1.0.0')).
+          to.equal('mongosh-1.0.0-darwin.zip');
+      });
+    });
+
+    context('when the build variant is linux', () => {
+      it('returns the tarball name', () => {
+        expect(tarballPath('', BuildVariant.Linux, '1.0.0')).
+          to.equal('mongosh-1.0.0-linux.tgz');
+      });
+    });
+
+    context('when the build variant is debian', () => {
+      it('returns the tarball name', () => {
+        expect(tarballPath('', BuildVariant.Debian, '1.0.0')).
+          to.equal('mongosh_1.0.0_amd64.deb');
+      });
+    });
+  });
+
+  describe('.tarballPosix', () => {
+    const version = '1.0.0';
+    const expectedTarball = tarballPath(__dirname, BuildVariant.Linux, version);
+
+    before(() => {
+      return tarballPosix(__dirname, expectedTarball);
+    });
+
+    after((done) => {
+      fs.unlink(expectedTarball, done);
+    });
+
+    it('builds the executable', (done) => {
+      fs.stat(expectedTarball, (error) => {
+        expect(error).to.equal(null);
+        done();
+      });
+    });
+  });
+
+  // macos and windows build variants does not come with installed 'dpkg' so
+  // this test fails. Run this test only on linux.
+  if (os.platform() === BuildVariant.Linux) {
+    describe('.tarballDebian', () => {
+      const version = '1.0.0';
+      const inputFile = path.join(__dirname, '..', 'examples', 'input.js');
+      const expectedTarball = tarballPath(__dirname, BuildVariant.Debian, version);
+
+      before(() => {
+        return tarballDebian(inputFile, __dirname, version, path.join(__dirname, '../../..'));
+      });
+
+      after((done) => {
+        fs.unlink(expectedTarball, done);
+      });
+
+      it('builds the executable', (done) => {
+        fs.stat(expectedTarball, (error) => {
+          expect(error).to.equal(null);
+          done();
+        });
+      });
+    });
+  }
+
+  describe('.tarballWindows', () => {
+    const version = '1.0.0';
+    const expectedTarball = tarballPath(__dirname, BuildVariant.Windows, version);
+    const inputFile = path.join(__dirname, '..', 'examples', 'input.js');
+
+    before(() => {
+      return tarballWindows(inputFile, expectedTarball);
+    });
+
+    after((done) => {
+      fs.unlink(expectedTarball, done);
+    });
+
+    it('builds the executable', (done) => {
+      fs.stat(expectedTarball, (error) => {
+        expect(error).to.equal(null);
+        done();
+      });
+    });
+  });
+
+  describe('.tarball', () => {
+    const platform = os.platform();
+    const version = '1.0.0';
+    const expectedTarball = tarballPath(__dirname, platform, version);
+    const inputFile = path.join(__dirname, '..', 'examples', 'input.js');
+    const rootDir = path.join(__dirname, '../../..');
+
+    before(() => {
```

Comment:
super minor: there is a bit of mix different async styles here, where possible it would be great if we can stick to one across the project

Reply:
ooooo you know I just did a `mv` on this file, and didn't actually look through previous tests that were here. Will change it up to what we are doing everywhere else.

Reply:
did you mean for something like this instead:
```js

  describe('.tarball', () => {
    const buildVariant = process.env.BUILD_VARIANT; 
    const version = '1.0.0';
    const expectedTarball = tarballPath(__dirname, buildVariant, version);
    const inputFile = path.join(__dirname, '..', 'examples', 'input.js');
    const rootDir = path.join(__dirname, '../../..');

    after(async () =>{
      await fs.unlink(expectedTarball);
    });

    it('builds the executable', async () => {
      await createTarball(inputFile, __dirname, buildVariant, version, rootDir);

      let accessErr;
      try {
        fs.access(expectedTarball);
      } catch (err) {
        accessErr = err;
      }

      expect(accessErr).to.equal(null)
    });
  });
```